File: GildedRoseInCpp/Weapon.cpp
#include "Weapon.h"
#include <stdexcept>
#include <sstream>

using std::stringstream;
using std::logic_error;
using std::endl;

void Weapon::AddAdditionalArticleInformation(stringstream& stringstream) const
{
	stringstream << "Damage Type: " << DamageTypeToString(_damageType) << endl;
	stringstream << "Minimum damage: " << _minimumDamage << endl;
	stringstream << "Maximum damage: " << _maximumDamage << endl;
	stringstream << "Hilt Type: " << HiltTypeToString(_hiltType) << endl;
}

Weapon::Weapon(std::string name, int quality, int durability, IUpdateArticleStrategy& strategy, int minimumDamage, int maximumDamage, DamageType damageType, HiltType hiltType)
	: IArticle(name, quality, durability, strategy), _minimumDamage(minimumDamage), _maximumDamage(maximumDamage), _damageType(damageType), _hiltType(hiltType)
{
	if (minimumDamage > maximumDamage)
		throw logic_error("minimumDamage cannot be greater than maximum damage");
}

File: GildedRoseDataBinding/Model/Weapon.cs
using System.Text;

namespace GildedRoseDataBinding.Model
{
    public class Weapon : Article
    {
        public DamageType DamageType { get; set; }
        public HiltType HiltType { get; set; }

        public int MinimumDamage { get; set; }
        public int MaximumDamage { get; set; }

        public Weapon(IUpdateArticleStrategy strategie)
            : base(strategie)
        {
            ArticleGroup = "Weapons";
        }

        protected override void AddSpecificArticleInfo(StringBuilder stringBuilder)
        {
            stringBuilder.AppendLine("Damage type: " + DamageType);
            stringBuilder.AppendLine(string.Format("Damage: {0} - {1}", MinimumDamage, MaximumDamage));
            stringBuilder.AppendLine("Hilt type: " + HiltType);
        }
    }
}